docs/TECHNICAL_DESIGN.md

```diff
@@ -0,0 +1,155 @@
+## Technical Design
+
+Carrot follows the Facebook Flux pattern:
+
+<img src="./flux-diagram-white-background.png" style="width: 100%;" />
+
+_Image source: the [Flux repo](https://github.com/facebook/flux)_
+
+It's important to note, Flux isn't a particular software library, instead it's an architectural pattern that's been implemented now in countless React apps and in many React web frameworks.
+
+The key idea in Flux is having a unidirectional data flow:
+
+<img src="./flux-simple-f8-diagram-with-client-action-1300w.png" style="width: 100%;" />
+
+_Image source: [Flux concepts](https://github.com/facebook/flux/tree/master/examples/flux-concepts)_
+
+Here is some background information on Flux:
+
+- https://github.com/facebook/flux - The Flux repo (documentation of the pattern, as well as a JavaScript Dispatcher implemntation)
+- https://github.com/facebook/flux/tree/master/examples/flux-concepts - High-level Flux concepts
+- https://facebook.github.io/flux/docs/in-depth-overview.html - Flux in depth
+- https://github.com/kgann/cljs-flux - An example implementation of the Flux pattern in ClojureScript
+
+### Action Creators
+
+An action creator is called by a user interaction or an API request finishing.
+
+These action creators are located in `/src/oc/web/actions/`. They are losely
+organized based the data they will act on. So `/src/oc/web/actions/user.cljs`
+contains actions for a user to sign up, login, and change user profile
+information.
+
```

Comment:
@thepug IMHO it would be worthwhile to have a code snippet of a typical action creator pair... maybe `query` and `query-finished` or `login-with-email` and `login-with-email-finish`.

Reply:
Alternatively, we could just make up examples. Maybe a simple action that doesn't need API and one that's paired with a finish.